pkg/daemon/ceph/client/pool.go

```diff
@@ -438,27 +442,30 @@ func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
 		}
 	}
 
-	args := []string{"osd", "pool", "create", pool.Name, pgCount, "replicated", crushRuleName, "--size", strconv.FormatUint(uint64(pool.Replicated.Size), 10)}
-	output, err := NewCephCommand(context, clusterInfo, args).Run()
+	poolDetails, err := GetPoolDetails(context, clusterInfo, pool.Name)
 	if err != nil {
```

Comment:
Can we be more precise and catch the error code instead of just a non-nil error?

Reply:
Yes I was thinking about that too, will check for a better error...

Reply:
After looking at this again, I think it's best if we still call `ceph osd pool create` even if there was another error besides the pool not existing. If the pool already exists, the pool create will succeed anyway, and if there is some other issue with the cluster, the pool create will fail the reconcile anyway.